include/libopencmsis/core_cm3.h

```diff
@@ -38,14 +38,27 @@
  * DeviceSupport/EnergyMicro/EFM32/efm32tg840f32.h, data from
  * libopencm3/cm3/scb.h. FIXME incomplete. */
 typedef struct {
-	__IO uint32_t CPUID;
+	__I  uint32_t CPUID;
 	__IO uint32_t ICSR;
 	__IO uint32_t VTOR;
 	__IO uint32_t AIRCR;
 	__IO uint32_t SCR;
 	__IO uint32_t CCR;
-	__IO uint8_t SHPR[12]; /* FIXME: how is this properly indexed? */
+	__IO uint8_t  SHPR[12]; /* FIXME: how is this properly indexed? */
 	__IO uint32_t SHCSR;
+	__IO uint32_t CFSR;
+	__IO uint32_t HFSR;
+	__IO uint32_t DFSR;
+	__IO uint32_t MMFAR;
+	__IO uint32_t BFAR;
+	__IO uint32_t AFSR;
+	__I  uint32_t PFR[2];
+	__I  uint32_t DFR;
+	__I  uint32_t ADR;
+	__I  uint32_t MMFR[4];
+	__I  uint32_t ISAR[5];
+		 uint32_t RESERVED0[5];
+	__IO uint32_t CPACR;
```

Comment:
aall of this is out of scope for this commit

Reply:
i will just undo all the changes in the core_cm3.h then.
i played a bit with debug stuff and wanted to look at the working register, which should be defined in this struct..